src/main/python/deep_qa/models/multiple_choice_qa/question_answer_memory_network.py

```diff
@@ -53,6 +53,6 @@ def _get_entailment_output(self, combined_input):
         # for a closer re-implementation of prior memory networks.
         answer_input = Input(shape=(self.num_options, self.max_answer_length), dtype='int32', name="answer_input")
         answer_embedding = self._embed_input(answer_input)
-        answer_encoder = EncoderWrapper(self._get_new_sentence_encoder(), name="answer_encoder")
+        answer_encoder = EncoderWrapper(self._get_encoder(), name="answer_encoder")
```

Comment:
Almost.  I would do `self._get_encoder(name="answer")`, or even `self._get_encoder(name=self.answer_encoder_name)`, where you add a parameter to the class that allows specifying the key to use in the dictionary for the answer encoder.  That would let you either use a new encoder or share the encoder, depending on what you want to do.